volttron/platform/control.py

```diff
@@ -439,42 +431,58 @@ def install_agent(self, filename, channel_name, vip_identity=None,
             Encoded public key the installed agent will use
         :param:string:secretkey:
             Encoded secret key the installed agent will use
+        :param:string:force:
+            Boolean value specifying whether the existence of an identity should
+            reinstall or cause an error.
         """
 
-        peer = bytes(self.vip.rpc.context.vip_message.peer).decode("utf-8")
+        # at this point if agent_uuid is populated then there is an
+        # identity of that already available.
+        agent_uuid = None
+        if vip_identity:
+            agent_uuid = self._identity_exists(vip_identity)
+        agent_existed_before = False
+        if agent_uuid:
+            agent_existed_before = True
+            if not force:
+                raise ValueError("Identity already exists, but not forced!")
```

Comment:
I get the same message with or without using --force

(volttron) volttron@volttron1:~/git/myvolttron$ vctl status
  AGENT             IDENTITY            TAG STATUS          HEALTH
a listeneragent-3.3 listeneragent-3.3_2
8 listeneragent-3.3 listeneragent-3.3_1
(volttron) volttron@volttron1:~/git/myvolttron$ vctl install examples/ListenerAgent --vip-identity listeneragent-3.3_2
install: error: Agent with VIP ID listeneragent-3.3_2 already installed on platform.
(volttron) volttron@volttron1:~/git/myvolttron$ vctl install examples/ListenerAgent --vip-identity listeneragent-3.3_2 --force
install: error: Agent with VIP ID listeneragent-3.3_2 already installed on platform.

Reply:
Oh interesting...my test starts the agents and that's why it passes.  It uses status_agents on the controlservice side rather than the list agent to determine if an identity exists or not.  This is a bug for sure.